sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClientAsync.md

```diff
@@ -0,0 +1,176 @@
+
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll-free phone plans can be found in the plan.
+
+```C# Snippet:TollFreePlanAreaCodesAsync
+var phonePlans = client.GetPhonePlansAsync(countryCode, tollFreePhonePlanGroup.PhonePlanGroupId, locale);
+var tollFreePhonePlan = (await phonePlans.ToEnumerableAsync()).First();
+
+foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+{
+    Console.WriteLine($"Area code: {areaCode}");
+}
+```
+
+## Reserve phone numbers
+
+Phone numbers need to be reserved before they can be purchased. Reservation is a long running operation that can be started by `CreateReservationOptions` function that returns an `PhoneNumberReservationOperation` object. `PhoneNumberReservationOperation` can be used to update status of the operation and to check for completeness.
+
+```C# Snippet:ReservePhoneNumbersAsync
+var reservationName = "My reservation";
+var reservationDescription = "reservation description";
+var reservationOptions = new CreateReservationOptions(reservationName, reservationDescription, new[] { phonePlanId }, areaCode);
+reservationOptions.Quantity = 1;
+
+var reserveOperation = await client.StartReservationAsync(reservationOptions);
+await reserveOperation.WaitForCompletionAsync();
+```
+
+## Persist reserve phone numbers operation
+
+You can persist the operation Id of the phone number reservation so that you can come back and check the operation status later.
+
+```C# Snippet:PersistReservePhoneNumbersOperationAsync
+var reservationId = reserveOperation.Id;
+
+// persist reservationOperationId and then continue with new operation
+
+var newReserveOperation = new PhoneNumberReservationOperation(client, reservationId);
+await newReserveOperation.WaitForCompletionAsync();
+```
+
+## Purchase phone numbers
+
+Phone numbers can be acquired through purchasing a reservation.
+
+```C# Snippet:StartPurchaseReservationAsync
+var reservationPurchaseOperation = await client.StartPurchaseReservationAsync(reservationId);
+await reservationPurchaseOperation.WaitForCompletionAsync();
+```
+
+## Listing acquired phone numbers
+
+You can list all phone numbers that have been acquired for your resource.
+
+```C# Snippet:ListAcquiredPhoneNumbersAsync
+var acquiredPhoneNumbers = client.GetAllPhoneNumbersAsync(locale);
+
+await foreach (var phoneNumber in acquiredPhoneNumbers)
+{
+    Console.WriteLine($"Phone number: {phoneNumber.PhoneNumber}, activation state: {phoneNumber.ActivationState}");
+}
+```
+
+## Release phone numbers
+
+If you no longer need a phone number you can release it.
+
+```C# Snippet:ReleasePhoneNumbersAsync
+var acquiredPhoneNumber = "<acquired_phone_number>";
+var phoneNumberReleaseOperation = client.StartReleasePhoneNumber(new PhoneNumber(acquiredPhoneNumber));
```

Comment:
Let's shorten the name to `releaseOperation`.

Reply:
Done